netx/net/sourceforge/jnlp/runtime/JNLPProxySelector.java

```diff
@@ -180,36 +199,95 @@ public void connectFailed(URI uri, SocketAddress sa, IOException ioe) {
      */
     @Override
     public List<Proxy> select(URI uri) {
-        OutputController.getLogger().log("Selecting proxy for: " + uri);
-        
-        if (inBypassList(uri)) {
-            List<Proxy> proxies = Arrays.asList(new Proxy[] { Proxy.NO_PROXY });
-            OutputController.getLogger().log("Selected proxies: " + Arrays.toString(proxies.toArray()));
-            return proxies;
-        }
+//        OutputController.getLogger().log("Selecting proxy for: " + uri);
+        OutputController.getLogger().log(OutputController.Level.MESSAGE_DEBUG, "Selecting proxy for: " + uri);
+
+
+
+
+//        if (inBypassList(uri)) {
+//            List<Proxy> proxies = Arrays.asList(new Proxy[] { Proxy.NO_PROXY });
+//            OutputController.getLogger().log(OutputController.Level.MESSAGE_DEBUG, "Selected proxies: " + Arrays.toString(proxies.toArray()));
+//            return proxies;
+//        }
 
         List<Proxy> proxies = new ArrayList<>();
 
         switch (proxyType) {
+            case PROXY_TYPE_BROWSER:
+                OutputController.getLogger().log(OutputController.Level.MESSAGE_DEBUG, "Configuring from Browser...");
+                String autoConfigString = "";
+                String proxyServer = "";
+                String proxyOver = "";
+                try {
+                    proxyServer = WinRegistry.readString(
+                            WinRegistry.HKEY_CURRENT_USER,
+                            PROXY_REGISTRY_KEY,
+                            PROXY_SERVER_REGISTRY_VAL, 0);
+
+                    if ((proxyServer.startsWith("ftp=")) || (proxyServer.startsWith("http=")) || (proxyServer.startsWith("https=")) || (proxyServer.startsWith("socks="))) {
+                        proxyServer = proxyServer.replaceAll("\\.*.*=" , "").replaceAll(";.*" , "");
+                    }
+                    else {
+                        proxyServer = proxyServer;
+                    }
+                    proxyOver = WinRegistry.readString(
+                            WinRegistry.HKEY_CURRENT_USER,
+                            PROXY_REGISTRY_KEY,
+                            PROXY_SERVER_OVERRIDE_VAL, 0);
+                    autoConfigString = WinRegistry.readString(
+                            WinRegistry.HKEY_CURRENT_USER,
+                            PROXY_REGISTRY_KEY,
+                            PROXY_AUTOCONFIG_URL_VAL, 0);
+                    if (proxyServer.equalsIgnoreCase("localhost:1234")) {
```

Comment:
What kind of convention is this?

Reply:
@judovana

this ist fake Proxy in our environment, I must check this value (localhost:1234).
you can drop it.